rewrite-java-11/src/main/java/org/openrewrite/java/isolated/ReloadableJava11ParserVisitor.java

```diff
@@ -1232,10 +1208,55 @@ public J visitTypeCast(TypeCastTree node, Space fmt) {
 
     @Override
     public J visitAnnotatedType(AnnotatedTypeTree node, Space fmt) {
+        if (node.getUnderlyingType() instanceof ArrayTypeTree) {
```

Comment:
Update:
I'm looking into issue https://github.com/openrewrite/rewrite/issues/3453, and came across this PR.

The `AnnotatedTypeTree` uses a `pos` field to retain where the annotation exists on the type.
Annotations that come before the `J tree` are added to the `annotations` field in the `J.AnnotatedType`.

AFAICT, `visitAnnotatedArrayType` will return a `J.ArrayType`, which does not have a container for leading annotations. So, this will cause an idempotent print issue for `@LeadingAnn Integer @TypeUsePosAnn [] ints`

Reply:
We could add new fields if that helps and even delete old ones if we deal with the deserialization compatibility. I would have to take a look again to figure it out, but IIRC we have a nested structure in one place and on the other hand we just have an int reflecting the number of dimensions. To support the annotations we should be using a nested structure.